src/FlaUILibrary/keywords/screenshot.py

```diff
@@ -17,19 +17,58 @@ def __init__(self, screenshots: Screenshot):
         self._screenshots = screenshots
 
     @keyword
-    def take_screenshot(self):
-        """ Takes a screenshot of the whole desktop. Returns path to the screenshot if screenshot was created.
+    def take_screenshot(self, base64 = False):
+        """ Takes a screenshot of the whole desktop. Returns path of screenshot created if ``base64`` is False (default).
+        Otherwise returns base64 encoded string of screenshot image.
+
+        Arguments:
+        | Argument      | Type   | Description          |
+        | base64        | string | True or False        |
 
         Example:
         | Take Screenshot |
         """
-        filepath = self._screenshots.execute_action(Screenshot.Action.CAPTURE,
-                                                    Screenshot.create_value_container())
+        image_var = None
+
+        if is_truthy(base64):
+            image_var = self._screenshots.execute_action(Screenshot.Action.CAPTURE_BASE64,
+                                                            Screenshot.create_value_container())
+            if image_var:
+                robotlog.log_screenshot_base64(image_var)
+        else:
+            image_var = self._screenshots.execute_action(Screenshot.Action.CAPTURE,
+                                                        Screenshot.create_value_container())
+            if image_var:
+                robotlog.log_screenshot(image_var)
+
+        return image_var
+
+    @keyword
+    def take_screenshot_from_element(self, element, base64 = False):
+        """ Takes a screenshot of the identified application window. Returns path of screenshot created.
+        
+        Arguments:
+        | Argument      | Type   | Description          |
+        | element       | string | XPath identifier from element |
+        | base64        | string | True or False        |
+
+        Example:
+        | Take Screenshot From Element   <XPATH>|
+        """
+        image_var = None
 
-        if filepath:
-            robotlog.log_screenshot(filepath)
+        if is_truthy(base64):
+            image_var = self._screenshots.execute_action(Screenshot.Action.CAPTURE_ELEMENT_BASE64,

```

Comment:
Please extend screenshot constructor with --> container: AutomationInterfaceContainer

AutomationInterfaceContainer contains an implementation to GetElementFromXpath

        module = self._container.create_or_get_module()
        element = module.get_element(identifier, None, msg=msg)

And set this element as input by given value container to receive a clean error handling without direct usage from C# code.

An example can be seen here --> 

https://github.com/GDATASoftwareAG/robotframework-flaui/blob/main/src/FlaUILibrary/keywords/grid.py#L18

https://github.com/GDATASoftwareAG/robotframework-flaui/blob/main/src/FlaUILibrary/keywords/grid.py#L46

For your use case InterfaceType is not required and can be set to Null.

By logic if xpath is Null or Empty code can be skipped and default handling can be used.

Reply:
done